Source/Core/Common/FileUtil.cpp

```diff
@@ -932,6 +1002,19 @@ static void RebuildUserDirectories(unsigned int dir_index)
         s_user_paths[D_CONFIG_IDX] + RETROACHIEVEMENTS_CONFIG;
     break;
 
+  case D_APPLICATIONSTATE_IDX:
+    s_user_paths[F_QSETTINGSCONFIG_IDX] = s_user_paths[D_APPLICATIONSTATE_IDX] + QSETTINGS_CONFIG;
+    s_user_paths[D_LOGS_IDX] = s_user_paths[D_APPLICATIONSTATE_IDX] + LOGS_DIR DIR_SEP;
+    s_user_paths[F_MAINLOG_IDX] = s_user_paths[D_LOGS_IDX] + MAIN_LOG;
+    s_user_paths[D_MAILLOGS_IDX] = s_user_paths[D_LOGS_IDX] + MAIL_LOGS_DIR DIR_SEP;
+    // The logs have moved to the state directory, so move the old ones.
+    // TODO: remove that someday.
+    MigrateDirectory(s_user_paths[D_USER_IDX] + LOGS_DIR DIR_SEP, s_user_paths[D_LOGS_IDX]);
+    // The QSettings config file has moved to the state directory, so move the old one.
+    // TODO: remove that someday.
+    MigrateFile(s_user_paths[D_CONFIG_IDX] + QSETTINGS_CONFIG, s_user_paths[F_QSETTINGSCONFIG_IDX]);
```

Comment:
Maybe we should only use the new locations if the old locations don't exist? Deleting the files from the old locations will break downgrades.